packages/office-ui-fabric-react/src/components/Button/CommandBarButton/CommandBarButton.styles.ts

```diff
@@ -1,57 +1,45 @@
-import { IButtonStyles } from '../Button.types';
-import {
-  ITheme,
-  concatStyleSets
-} from '../../../Styling';
-import { memoizeFunction } from '../../../Utilities';
-import {
-  getStyles as getBaseButtonStyles
-} from '../BaseButton.styles';
-import {
-  getStyles as getSplitButtonStyles
-} from '../SplitButton/SplitButton.styles';
-
-export const getStyles = memoizeFunction((
-  theme: ITheme,
-  customStyles?: IButtonStyles,
-  focusInset?: string,
-  focusColor?: string
-): IButtonStyles => {
-  let baseButtonStyles: IButtonStyles = getBaseButtonStyles(theme);
-  let baseSplitButtonStyles: IButtonStyles = getSplitButtonStyles(theme);
-  let commandButtonStyles: IButtonStyles = {
-    root: {
-      minWidth: '40px',
-      backgroundColor: theme.palette.neutralLighter,
-      color: theme.palette.neutralPrimary,
-      padding: '0 4px'
-    },
-
-    rootHovered: {
-      backgroundColor: theme.palette.neutralLight,
-      color: theme.palette.neutralDark
-    },
-
-    rootPressed: {
-      backgroundColor: theme.palette.neutralQuaternaryAlt,
-      color: theme.palette.black
-    },
-
-    rootChecked: {
-      backgroundColor: theme.palette.neutralQuaternaryAlt,
-      color: theme.palette.black
-    },
-
-    rootExpanded: {
-      backgroundColor: theme.palette.neutralQuaternaryAlt,
-      color: theme.palette.black
-    },
-
-    rootCheckedHovered: {
-      backgroundColor: theme.palette.neutralQuaternary,
-      color: theme.palette.black,
-    },
-
+import { concatStyleSets, FontWeights } from '../../../Styling';
+import { IButtonBaseStyleProps, IButtonBaseStyles } from '../_base/Button.base.types';
+
+import { getButtonBaseStyles } from '../_base/Button.base.styles';
+
+export const getStyles = (props: IButtonBaseStyleProps): IButtonBaseStyles => {
+  const { theme, checked, disabled, expanded } = props;
+
+  let buttonStyles: IButtonBaseStyles = {
+    button: [
+      'ms-CommandBarButton',
+      !disabled && {
+        minWidth: 40,
+        backgroundColor: theme.palette.neutralLighter,
```

Comment:
At some point, are we going to move away from the palette colors and move towards semantic slots for this kind of thsing? (obviously outside the scope of this PR)